framework/src/base/FEProblem.C

```diff
@@ -3582,11 +3573,10 @@ FEProblem::adaptMesh()
   unsigned int cycles_per_step = _adaptivity.getCyclesPerStep();
   for (unsigned int i=0; i < cycles_per_step; ++i)
   {
-    _console << "Adaptivity step " << i+1 << " of " << cycles_per_step << "\n";
+    _console << "Adaptivity step " << i+1 << " of " << cycles_per_step << std::endl;
```

Comment:
After looking at this again, the std::endl; is needed. The printing of the
mesh information that happens in Console::meshChanged() goes through
Console::mooseConsole. This method applies the MultiApp indent and then
uses Moose::out. Therefore, it skips the _console buffer. Thus, without the
std::endl the order is incorrect.

Perhaps, this isn't the correct way to handle this, but within the Console
object I avoid using _console because this adds a bunch of routing just to
come back to the same object. Perhaps, the correct solution is that
Console::mooseConsole should flush the buffer so we don't need to worry
about it.

On Thu, Apr 16, 2015 at 10:47 AM, Cody Permann <notifications@github.com>
wrote:

> In framework/src/base/FEProblem.C
> <https://github.com/idaholab/moose/pull/4920#discussion_r28527257>:
>
> > @@ -3582,11 +3573,10 @@ FEProblem::adaptMesh()
> >    unsigned int cycles_per_step = _adaptivity.getCyclesPerStep();
> >    for (unsigned int i=0; i < cycles_per_step; ++i)
> >    {
> > -    _console << "Adaptivity step " << i+1 << " of " << cycles_per_step << "\n";
> > +    _console << "Adaptivity step " << i+1 << " of " << cycles_per_step << std::endl;
>
> I disagree, that's exactly what a buffer is for, buffering output. Even
> though this branch of code is rarely called and it's not a point of
> slowness I think we should clean it up for consistency.
>
> We should strive to cleanup the framework so that we flush when we need to
> see output and use normal newline characters where we don't. As long as
> meshChanged is printing to the console there is not need to flush here. If
> anything the flush should come "after" the mesh information is printed not
> after the header and before the body. I have removed dozens of unnecessary
> std::endls throughout the code, let's keep that going.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/4920/files#r28527257>.
>